### ISSUE TITLE
Not mounted cgroup blocks docker container start

### ISSUE THREAD

**Issue by Snorch** [moby/moby#37601]
**Description**

When I create my own empty named cgroup hierarchy on the host where docker is installed, for instance for grouping processes, and later umount it while it still contains some non-root cgroup directories, docker becomes unusable.

**Steps to reproduce the issue:**

```
[root@snorch snorch]# cat docker_test.sh
#!/bin/bash
mount -t cgroup -o none,name=test cgroup /tmp/test
mkdir /tmp/test/dir
umount /tmp/test

# Wait for cgroup to be created
while :; do
	cat /proc/self/cgroup | grep test &>/dev/null && break
done

# Run docker with test cgroup having no mount
docker run --rm -t busybox echo Docker Started!

mount -t cgroup -o none,name=test cgroup /tmp/test
rmdir /tmp/test/dir
umount /tmp/test

# Wait for cgroup to destroy
while :; do
	cat /proc/self/cgroup | grep test &>/dev/null || break
done

# Run docker without test cgroup
docker run --rm -t busybox echo Docker Started!
```
```
[root@snorch snorch]# bash docker_test.sh
docker: Error response from daemon: cgroups: cannot find cgroup mount destination: unknown.
Docker Started!
```

**Describe the results you expected:**

```
[root@snorch snorch]# bash docker_test.sh
Docker Started!
Docker Started!
```

**Output of `docker version`:**

```
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:11:45 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:14:07 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Output of `docker info`:**

```
Containers: 38
 Running: 0
 Paused: 0
 Stopped: 38
Images: 57
Server Version: 18.06.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: d64c661f1d51c48782c9cec8fda7604785f93587
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.17.6-200.fc28.x86_64
Operating System: Fedora 28 (Workstation Edition)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.52GiB
Name: snorch
ID: FOBI:53SZ:3XT7:5E4Q:JJND:QPOY:PQAT:GPCK:IVQ6:L2AO:IZPY:QWKU
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Happens on my physical node, criu tests create similar cgroups when run, that's why I mentioned these issue.

**Comment by Snorch** [comment]
Just to clarify, fix to criu(zdtm) removes leftover cgroups after test finishes, so that we can still run docker after test. But we still need to handle such a cgroup configuration in docker properly.

**Comment by thaJeztah** [comment]
closing, as this went stale

**Comment by Snorch** [comment]
This seems to be fixed on `Docker version 25.0.3, build 4debf41`, likely after multiple cgroup-v2 related changes.
